sharding/contracts/sharding_manager.sol

```diff
@@ -1,253 +1,228 @@
+    struct HeaderVars {
+        bytes32 entireHeaderHash;
+        int score;
+        address notaryAddr;
+        bool isNewHead;
+    }
+
+    address[] public notaryPool;
+
+    // notaryAddress => notaryStruct
+    mapping (address => Notary) public notaryRegistry;
+    // shardId => (headerHash => treeRoot)
+    mapping (uint => mapping (bytes32 => bytes32)) public collationTrees;
+    // shardId => (headerHash => collationHeader)
+    mapping (int => mapping (bytes32 => CollationHeader)) public collationHeaders;
+    // shardId => headerHash
+    mapping (int => bytes32) shardHead;
+
+    // Number of notaries
+    uint public notaryPoolLength;
+    // Stack of empty notary slot indicies
+    uint[] emptySlotsStack;
+    // Top index of the stack
+    uint emptySlotsStackTop;
+
+    // Notary sample size at current period and next period
+    uint currentPeriodNotarySampleSize;
+    uint nextPeriodNotarySampleSize;
+    uint sampleSizeLastUpdatedPeriod;
+
+    // Constant values
+    uint constant PERIOD_LENGTH = 5;
+    // Number of shards
+    uint constant SHARD_COUNT = 100;
+    // The minimum deposit size for a notary
+    uint constant NOTARY_DEPOSIT = 1000 ether;
+    // The reward for notary on voting for a collation
+    uint constant NOTARY_REWARD = 0.001 ether;
+    // Time the ether is locked by notaries
+    uint constant NOTARY_LOCKUP_LENGTH = 16128;
+    // Number of periods ahead of current period, which the contract
+    // is able to return the notary of that period
+    uint constant LOOKAHEAD_LENGTH = 4;
+    // Number of notaries to select from notary pool for each shard in each period
+    uint constant COMMITTEE_SIZE = 135;
+    // Threshold(number of notaries in committee) for a proposal to be accepted
+    uint constant QUORUM_SIZE = 90;
+
+    // Log the latest period number of the shard
+    mapping (int => int) public periodHead;
+
+    /// Checks if a notary with given shard id and period has been chosen as
+    /// a committee member to vote for header added on to the main chain
+    function getNotaryInCommittee(uint shardId, uint _index) public view returns(address) {
+        uint period = block.number / PERIOD_LENGTH;
+
+        // Determine notary pool length based on notary sample size
+        uint sampleSize;
+        if (period > sampleSizeLastUpdatedPeriod) {
+            sampleSize = nextPeriodNotarySampleSize;
+        } else {
+            sampleSize = currentPeriodNotarySampleSize;
+        }
+
+      // Get the most recent block number before the period started
+        uint latestBlock = period * PERIOD_LENGTH - 1;
+        uint latestBlockHash = uint(block.blockhash(latestBlock));
+        uint index = uint(keccak256(latestBlockHash, _index, shardId)) % sampleSize;
+
+        return notaryPool[index];
+    }
+
+    /// Registers notary to notatery registry, locks in the notary deposit,
+    /// and returns true on success
+    function registerNotary() public payable returns(bool) {
```

Comment:
Why do we return a boolean? Wouldn't the transaction fail if there was an error? 
Is this standard practice for solidity?

Reply:
I'm following the latest spec and py-evm's implementation: https://ethresear.ch/t/minimal-sharding-protocol-extension/1823. We can remove it if there's a strong consensus of doing so

Reply:
Let's remove it if we aren't doing anything with it.